src/rdkafka.h

```diff
@@ -2957,6 +2973,22 @@ rd_kafka_position (rd_kafka_t *rk,
  * `rd_kafka_topic_new()`.
  *
  * `rd_kafka_produce()` is an asynch non-blocking API.
+ * See `rd_kafka_conf_set_dr_msg_cb` on how to setup a callback to be called
+ * once the delivery status (success or failure) is known.
+ *
+ * Since producing is asynchronous, you should call `rd_kafka_flush()` before
+ * you destroy the producer. Otherwise, any outstanding messages will be
+ * discarded and you won't know. An alternative is to check for delivery
+ * reports via `rd_kafka_conf_set_dr_msg_cb()`.
+ *
+ * When errors occur, librdkafka automatically retries to produce the
+ * messages. Retries are triggered after retry.backoff.ms but only if the
+ * leader broker for the given partition is available. Otherwise, librdkafka
+ * falls back to polling the topic metadata to monitor when a new leader is
+ * elected (see the topic.metadata.refresh.fast.interval.ms and
+ * topic.metadata.refresh.interval.ms configurations) and then performs a
+ * retry. A delivery error will always occur if overall it took more than
```

Comment:
change last line to "A delivery error will occur if the message could not be produced within `message.timeout.ms`."